src/data/equiv/module.lean

```diff
@@ -477,3 +477,31 @@ def to_module_aut : S →* M ≃ₗ[R] M :=
   map_mul' := λ a b, linear_equiv.ext $ mul_smul _ _ }
 
 end distrib_mul_action
+
+namespace add_equiv
+variables [semiring R] [add_comm_monoid M] [module R M] [add_comm_monoid M₂] [module R M₂]
+
+/-- An additive equivalence whose underlying function preserves `smul` is a linear equivalence. -/
+def to_linear_equiv (e : M ≃+ M₂) (h : ∀ (c : R) x, e (c • x) = c • e x) : M ≃ₗ[R] M₂ :=
+{ map_smul' := h, .. e, }
+
+@[simp] lemma coe_to_linear_equiv (e : M ≃+ M₂) (h : ∀ (c : R) x, e (c • x) = c • e x) :
+  ⇑(e.to_linear_equiv h) = e :=
+rfl
+
+@[simp] lemma coe_to_linear_equiv_symm (e : M ≃+ M₂) (h : ∀ (c : R) x, e (c • x) = c • e x) :
+  ⇑(e.to_linear_equiv h).symm = e.symm :=
+rfl
+
+/-- An additive equivalence between commutative groups is a linear equivalence between ℤ-modules -/
+def to_int_linear_equiv {α β : Type*} [add_comm_group α] [add_comm_group β]
+  (e : α ≃+ β) : α ≃ₗ[ℤ] β :=
+e.to_linear_equiv $ λ c a, by { erw e.to_add_monoid_hom.map_zsmul, refl }
```

Comment:
Could you add analogous `simp` lemmas for `to_int_linear_equiv` and `to_nat_linear_equiv`, similar to the ones for `coe_to_linear_equiv` and `coe_to_linear_eequiv_symm`? It's possible you can use `@[simps]` to automatically generate these lemmas (and you can check the lemmas so generated using `@[simps?]`).

Reply:
The `[@simps]` generated ones are eta-expanded, and thus weaker than the manual ones that we have for `coe_to_linear_equiv`:
```
[simps] > The given definition is not a constructor application:
        >   e.to_linear_equiv _
        > Retrying with the options { rhs_md := semireducible, simp_rhs := tt}.
[simps] > adding projection add_equiv.to_int_linear_equiv_apply:
        > ∀ {α : Type u_1} {β : Type u_2} [_inst_6 : add_comm_group α] [_inst_7 : add_comm_group β]
(e : α ≃+ β) (ᾰ : α), ⇑(e.to_int_linear_equiv) ᾰ = ⇑e ᾰ
[simps] > adding projection add_equiv.to_int_linear_equiv_symm_apply:
        > ∀ {α : Type u_1} {β : Type u_2} [_inst_6 : add_comm_group α] [_inst_7 : add_comm_group β]
(e : α ≃+ β) (ᾰ : β), ⇑(e.to_int_linear_equiv.symm) ᾰ = ⇑(e.symm) ᾰ
```
so I guess I’ll have to write them by hand as well

Reply:
Add the cargo-cult-copies of what we have for `coe_to_linear_equiv`, plus the one for `symm`. I am not sure what the desired formulation for `.refl` and `.trans` would be.